frontend/js/src/settings/brainzplayer/BrainzPlayerSettings.tsx

```diff
@@ -0,0 +1,216 @@
+  const [youtubeEnabled, setYoutubeEnabled] = React.useState(
+    userPreferences?.brainzplayer?.youtubeEnabled ?? true
+  );
+  const [spotifyEnabled, setSpotifyEnabled] = React.useState(
+    userPreferences?.brainzplayer?.spotifyEnabled ??
+      SpotifyPlayer.hasPermissions(spotifyAuth)
+  );
+  const [soundcloudEnabled, setSoundcloudEnabled] = React.useState(
+    userPreferences?.brainzplayer?.soundcloudEnabled ??
+      SoundcloudPlayer.hasPermissions(soundcloudAuth)
+  );
+
+  const saveSettings = React.useCallback(async () => {
+    if (!currentUser?.auth_token) {
+      toast.error("You must be logged in to update your preferences");
+      return;
+    }
+    const { submitBrainzplayerPreferences } = APIService;
+    try {
+      await submitBrainzplayerPreferences(currentUser.auth_token, {
+        youtubeEnabled,
+        spotifyEnabled,
+        soundcloudEnabled,
+      });
+      toast.success("Saved your preferences successfully");
+    } catch (error) {
+      toast.error(
+        <ToastMsg
+          title="Error saving preferences"
+          message={
+            <>
+              {error.toString()}
+              <br />
+              Please try again or contact us if the issue persists.
+            </>
+          }
+        />
+      );
+    }
+  }, [
+    youtubeEnabled,
+    spotifyEnabled,
+    soundcloudEnabled,
+    APIService,
+    currentUser?.auth_token,
+  ]);
+
+  return (
+    <>
+      <Helmet>
+        <title>BrainzPlayer Settings</title>
+      </Helmet>
+      <h2 className="page-title">BrainzPlayer Settings</h2>
+      <h4 className="page-title">Music Players</h4>
+      <p>
+        You can choose which music services to play music from on listenBrainz.
+        To sign in and authorize your account please go to the{" "}
+        <Link to="/settings/music-services/">music services page</Link>. <br />
+        If you don&apos;t have a subscription to a music service listed below,
+        Youtube will be used by default as it does not require an account.
+        However the search results from Youtube are often inaccurate, and for a
+        better listening experience we recommend one of the other services (if
+        you have one available).
+      </p>
+      <p>
+        <Switch
+          id="enable-spotify"
+          disabled={!SpotifyPlayer.hasPermissions(spotifyAuth)}
+          value="spotify"
+          checked={spotifyEnabled}
+          onChange={(e) => setSpotifyEnabled(!spotifyEnabled)}
+          switchLabel={
+            <span
+              className={`text-brand ${spotifyEnabled ? "text-success" : ""}`}
+            >
+              <FontAwesomeIcon icon={faSpotify} />
+              &nbsp;Spotify
+            </span>
+          }
+        />
+        <br />
+        <small>
+          Spotify requires a premium account to play full songs on third-party
+          websites. To sign in, please go to the{" "}
+          <Link to="/settings/music-services/">music services page</Link>.
```

Comment:
The link here is incorrect
```suggestion
          <Link to="/settings/music-services/details/">music services page</Link>.
```